pkg/store/proxy.go

```diff
@@ -81,38 +81,73 @@ func (s *ProxyStore) Info(ctx context.Context, r *storepb.InfoRequest) (*storepb
 		StoreType: s.component.ToProto(),
 	}
 
-	MinTime := int64(math.MaxInt64)
-	MaxTime := int64(0)
-
+	minTime := int64(math.MaxInt64)
+	maxTime := int64(0)
 	stores := s.stores()
+
+	if len(stores) == 0 {
+		res.MaxTime = math.MaxInt64
+		res.MinTime = 0
+
+		return res, nil
+	}
+
 	for _, s := range stores {
 		mint, maxt := s.TimeRange()
-		if mint < MinTime {
-			MinTime = mint
+		if mint < minTime {
+			minTime = mint
 		}
-		if maxt > MaxTime {
-			MaxTime = maxt
+		if maxt > maxTime {
+			maxTime = maxt
 		}
 	}
 
-	// Edge case: we have all of the data if there are no stores.
-	if len(stores) == 0 {
-		MinTime = 0
-		MaxTime = math.MaxInt64
-	}
-
-	res.MaxTime = MaxTime
-	res.MinTime = MinTime
+	res.MaxTime = maxTime
+	res.MinTime = minTime
 
 	for _, l := range s.selectorLabels {
 		res.Labels = append(res.Labels, storepb.Label{
 			Name:  l.Name,
 			Value: l.Value,
 		})
 	}
+
+	labelSets := make(map[uint64][]storepb.Label, len(stores))
+	for _, st := range stores {
+		for _, labelSet := range st.LabelSets() {
+			mergedLabelSet := mergeLabels(labelSet.Labels, s.selectorLabels)
+			ls := storepb.LabelsToPromLabels(mergedLabelSet)
+			sort.Sort(ls)
+			labelSets[ls.Hash()] = mergedLabelSet
+		}
+	}
+
+	res.LabelSets = make([]storepb.LabelSet, 0, len(labelSets))
+	for _, v := range labelSets {
+		res.LabelSets = append(res.LabelSets, storepb.LabelSet{Labels: v})
```

Comment:
Worth to mention on interface that sets are not sorted?

Reply:
Added to the proto message and generated structs.